src/families/bitcoin/js-synchronisation.ts

```diff
@@ -323,17 +323,18 @@ const getAccountShape: GetAccountShape = async (info) => {
   const { txs: transactions } = await wallet.getAccountTransactions(
     walletAccount
   );
+
+  const accountAddresses: Set<string> = new Set<string>();
   const accountAddressesWithInfo = await walletAccount.xpub.getXpubAddresses();
-  const accountAddresses = accountAddressesWithInfo
-    ? accountAddressesWithInfo.map((a) => a.address)
-    : [];
+  accountAddressesWithInfo.map((a) => accountAddresses.add(a.address));
+
+  const changeAddresses: Set<string> = new Set<string>();
   const changeAddressesWithInfo =
     await walletAccount.xpub.storage.getUniquesAddresses({
       account: 1,
     });
-  const changeAddresses = changeAddressesWithInfo
-    ? changeAddressesWithInfo.map((a) => a.address)
-    : [];
+  changeAddressesWithInfo.map((a) => changeAddresses.add(a.address));
```

Comment:
maybe
const changeAddresses = new Set( changeAddressesWithInfo.map(  a => a.address));
would be more concise.